README.md

```diff
@@ -17,6 +17,23 @@ Please chat with us on the [discussion board](https://github.com/ethereum-optimi
 
 We welcome your contributions. Read through [CONTRIBUTING.md](./CONTRIBUTING.md) for a general overview of the contributing process for this repository.
 
+### How to Think about Specs
+
+The specs process is a schelling point for where core developers go to modify the OP Stack. A spec serves multiple purposes:
+- Ensures that we know what we are building before we build it
+- Helps code reviewers and auditors understand what the behavior is supposed to be
+- Enables multiple implementations
+
+Some guidelines for writing a good spec:
```

Comment:
I think we should have more emphasis on the idea that specs should:
- Focus on describing the interfaces, behaviors, requirements, and safety conditions (e.g. when to error)
- Avoid mentioning implementation details and executable code snippets (e.g. interface code snippets are ok, implementation snippets are not).

This is an important distinction and leaking implementation details is an easy, expected mistake to make for people new to the spec-writing process. Quoting from a reply I left in https://github.com/ethereum-optimism/specs/pull/233#discussion_r1640544072:

> To expand a bit so I don't think we've discussed this much, the rationale for "not leaking implementation details into the spec" is:
> 
> - Simplify spec maintenance: By specifying as little as possible to implement the requirements, we make it easier to keep the spec up to date by having less code to maintain. The code in this spec is not tested or audited, and we may end up tweaking it.
> - Readability: The less code, the easier it is for a less-technical reader to understand the spec and ensure it's sound.
> - Testing: The implementer of the spec has flexibility around the final result, as long as it meets the requirements. When implementing the spec, we write tests for all the requirements in here, but we wouldn't write a test to e.g. ensure we're using assembly since that's an implementation detail and not a requirement.

Reply:
I agree with this point. Would be good to see future PRs add this kind of content.